pkg/instancegroups/instancegroups.go

```diff
@@ -387,7 +387,8 @@ func (c *RollingUpdateCluster) drainTerminateAndWait(u *cloudinstances.CloudInst
 }
 
 func (c *RollingUpdateCluster) reconcileInstanceGroup() error {
-	if api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderOpenstack {
+	if api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderOpenstack ||
+		api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderDO {
```

Comment:
Always true, maybe you wanted AND?
```suggestion
	if api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderOpenstack &&
		api.CloudProviderID(c.Cluster.Spec.CloudProvider) != api.CloudProviderDO {
```

Reply:
Good catch, thanks @hakman